python/delta_sharing/reader.py

```diff
@@ -71,19 +79,112 @@ def limit(self, limit: Optional[int]) -> "DeltaSharingReader":
             timestamp=self._timestamp
         )
 
-    def to_pandas(self) -> pd.DataFrame:
+    def _get_response(self) -> ListFilesInTableResponse:
         response = self._rest_client.list_files_in_table(
             self._table,
             predicateHints=self._predicateHints,
             limitHint=self._limit,
             version=self._version,
-            timestamp=self._timestamp
+            timestamp=self._timestamp,
         )
 
+        return response
+
+    @staticmethod
+    def _to_pyarrow_dataset(
+        file_urls: Union[Sequence[str], str], pyarrow_ds_options: Optional[Mapping[str, Any]] = None
+    ) -> PyArrowDataset:
+        if pyarrow_ds_options is None:
+            pyarrow_ds_options = {}
+
+        if file_urls and isinstance(file_urls, list):
+            first_url = urlparse(file_urls[0])
+        elif file_urls and isinstance(file_urls, str):
+            first_url = urlparse(file_urls)
+        else:
+            raise ValueError(
+                "PyArrow Dataset can only be instantiated from a path of a list of paths."
+            )
+
+        first_scheme = first_url.scheme
+
+        if "storage.googleapis.com" in (first_url.netloc.lower()):
+            # Apply the yarl patch for GCS pre-signed urls
+            import delta_sharing._yarl_patch  # noqa: F401
+
+        # Take URL of first file from response and assert that all
+        # the other URLs have same scheme as well.
+        # Otherwise, fsspec filesystem cannot be instantiated.
+        if file_urls and isinstance(file_urls, list):
+            same_scheme = True
+            for url in file_urls:
+                if not urlparse(url).scheme == first_scheme:
+                    same_scheme = False
+                    break
+
+            assert same_scheme, "All files did not follow the same URL scheme."
```

Comment:
nit: "All files should follow the same URL scheme" ?

Reply:
Question: what's an example of this failure? And is it possible to add a test case to cover it?

Reply:
I dont think the delta server can return files from different places for for the same table but I added this case just in case if in the future delta sharing turns in to a cross cloud service (some data in s3 and some data in GCS).

Another reason for adding this was that we dont have to initialize FSSPEC FS for each path if they all follow the same scheme.

Reply:
I don't foresee delta sharing support a single table from multiple clouds any time soon. 
Plus no test coverage on the code, could we rather turn this into a TODO.